src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java

```diff
@@ -59,38 +61,59 @@ void populateByDueDiligenceDto(
         dueDiligence.setDiligence(dueDiligenceDto.getDiligence());
     }
 
-    void populateByOEDueDiligenceDto(
-            DueDiligence dueDiligence, OverseasEntityDueDiligenceDto overseasEntityDueDiligenceDto) {
+    private void populateByOEDueDiligenceDto(
+            DueDiligence dueDiligence,
+            OverseasEntityDueDiligenceDto overseasEntityDueDiligenceDto) {
 
         dueDiligence.setDateChecked(overseasEntityDueDiligenceDto.getIdentityDate().toString());
         dueDiligence.setAgentName(overseasEntityDueDiligenceDto.getName());
-        dueDiligence.setDueDiligenceCorrespondenceAddress(overseasEntityDueDiligenceDto.getAddress());
+        dueDiligence.setDueDiligenceCorrespondenceAddress(
+                overseasEntityDueDiligenceDto.getAddress());
         dueDiligence.setSupervisoryBody(overseasEntityDueDiligenceDto.getSupervisoryName());
         dueDiligence.setPartnerName(overseasEntityDueDiligenceDto.getPartnerName());
         dueDiligence.setEmail(overseasEntityDueDiligenceDto.getEmail());
         dueDiligence.setAmlRegistrationNumber(overseasEntityDueDiligenceDto.getAmlNumber());
     }
 
-    void populateFilingForDate(
-            OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
-        var filingForDate = new FilingForDate();
+    private void populatePresenter(
+            OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+            UpdateSubmission updateSubmission) {
+        if (overseasEntitySubmissionDto.getPresenter() != null) {
```

Comment:
We really should have presenter always so either checks are redundant or we risk passing incomplete submission on.

Reply:
I've updated this and refactored to use Optionals